File: python/ai_convert.py
# Experimental: convert a plain PDF invoice to Factur-X using AI.
# Review output manually before sending to a customer or tax authority.
#
# Get API key: https://www.invoicexml.com/account/authentication
# Docs:        https://www.invoicexml.com/docs/api/transform/to/facturx
#
# Dependency:  pip install requests

import requests
import sys

api_key = "YOUR_API_KEY"

response = requests.post(
    "https://api.invoicexml.com/v1/transform/to/facturx",
    headers={"Authorization": f"Bearer {api_key}"},
    files={"file": ("plain-invoice.pdf", open("plain-invoice.pdf", "rb"), "application/pdf")},
    data={
        "version":  "2.3.2",
        "profile":  "extended",
        "syntax":   "cii",
        "embed":    "true",
        "language": "en",
    },
)

if not response.ok:
    sys.stderr.write(f"InvoiceXML API error {response.status_code}: {response.text}\n")
    sys.exit(1)

with open("converted-facturx.pdf", "wb") as f:
    f.write(response.content)

print(f"Saved converted-facturx.pdf ({len(response.content)} bytes)")
print("Review the output manually before sending.")

File: java/AiConvert.java
// Experimental: convert a plain PDF invoice to Factur-X using AI.
// Review output manually before sending to a customer or tax authority.
//
// Get API key: https://www.invoicexml.com/account/authentication
// Docs:        https://www.invoicexml.com/docs/api/transform/to/facturx
//
// Dependency: com.squareup.okhttp3:okhttp:4.12.0

import okhttp3.*;
import java.io.File;
import java.nio.file.Files;
import java.nio.file.Paths;

public class AiConvert {
    public static void main(String[] args) throws Exception {
        String apiKey = "YOUR_API_KEY";

        RequestBody body = new MultipartBody.Builder()
            .setType(MultipartBody.FORM)
            .addFormDataPart("file", "plain-invoice.pdf",
                RequestBody.create(new File("plain-invoice.pdf"), MediaType.parse("application/pdf")))
            .addFormDataPart("version",  "2.3.2")
            .addFormDataPart("profile",  "extended")
            .addFormDataPart("syntax",   "cii")
            .addFormDataPart("embed",    "true")
            .addFormDataPart("language", "en")
            .build();

        Request request = new Request.Builder()
            .url("https://api.invoicexml.com/v1/transform/to/facturx")
            .header("Authorization", "Bearer " + apiKey)
            .post(body)
            .build();

        try (Response response = new OkHttpClient().newCall(request).execute()) {
            byte[] pdf = response.body().bytes();
            if (!response.isSuccessful()) {
                System.err.println("InvoiceXML API error " + response.code() + ": " + new String(pdf));
                System.exit(1);
            }
            Files.write(Paths.get("converted-facturx.pdf"), pdf);
            System.out.println("Saved converted-facturx.pdf (" + pdf.length + " bytes)");
            System.out.println("Review the output manually before sending.");
        }
    }
}
